### PR TITLE
fix mypy

### LINKED ISSUE #28: Broken mypy checks
pytket 1.19.0 added various type hints. Now the tests in `pyzx_convert_test` fail the mypy check:

https://github.com/CQCL/pytket-pyzx/blob/9ab6d60625b8c67994ac7d64c10dca84069a3ed0/tests/pyzx_convert_test.py#L77

Constructing an `Architecture` requires a `list[tuple[int, int]]` instead of a `list[list[int]]`.

See https://github.com/CQCL/pytket-pyzx/actions/runs/6140422592/job/16659369142?pr=27

### PR DESCRIPTION
**Pull request by cqc-melf**
fix https://github.com/CQCL/pytket-pyzx/issues/28

### CODE REVIEW

File: pytket/extensions/pyzx/pyzx_convert.py [hunk 1]
```diff
@@ -229,7 +229,7 @@ def tk_to_pyzx_placed_circ(
         qubit_dict[q] = i
 
     for i, x in enumerate(pytket_arc.nodes):
-        arc_dict[x] = qubit_dict[q_map[x]]
+        arc_dict[x] = qubit_dict[q_map[x]]  # type: ignore
```
Comment (cqc-melf) [inline pytket/extensions/pyzx/pyzx_convert.py:232]: I have not found a solution without an ignore, happy to take any suggestion for this

**Comment by cqc-alec** [comment]
Could we also add `warn_unused_ignores = True` to `mypy.ini` to catch any unneeded ones? I've been doing this for the other extensions while fixing up the issues.

**Comment by cqc-melf** [comment]
> Could we also add `warn_unused_ignores = True` to `mypy.ini` to catch any unneeded ones? I've been doing this for the other extensions while fixing up the issues.

done

File: mypy.ini [hunk 2]
```diff
@@ -19,6 +19,8 @@ warn_no_return = False
 warn_return_any = True
 warn_unreachable = True
 
+warn_unused_ignores = True
```
Comment (cqc-alec) [inline mypy.ini]: Please also delete the line `warn_unused_ignores = False` above.

**Review by aborgna-q — APPROVED** [review]

**Pull request merged**